warehouse/oidc/models/github.py

```diff
@@ -189,7 +189,26 @@ def __lookup_all__(klass, signed_claims: SignedClaims) -> Query | None:
                 environment=environment.lower(),
             )
             .filter(
-                literal(workflow_ref).like(func.concat(klass.workflow_filename, "%"))
+                literal(workflow_ref).startswith(
```

Comment:
Since we're switching the use of `like(concat(name, %))` with `.startswith(name)`, we could go one step further and fix the issue we discussed where `filename.yml.whatever.yml` will match the TP for `filename.yml` (due to the `%` at the end).

 We could do `.startswith(name + '@')`, which would fix the issue for all cases except for the case where another `@` is used, like `filename.yml@whatever.yml`.

Reply:
Yeah, I was thinking a bit about that -- I think we can address both of those cases at the same time by imitating what we do for the `_check_job_workflow_ref` claim validator, i.e. *reconstruct* the parts of the `workflow_ref` rather than trying to disambiguate them. The main downside to that is that it means we'll have come full circle, and will essentially be doing claim validation twice (once at the DB level, once at the `verify_claims` level).

(The other alternative here would be to have GitHub fix this at the claims level, e.g. by adding a new `workflow_filename` or similar claim. I've pinged some folks to see if that's something they can do.)

Reply:
(And NB: the other benefit to a new `workflow_filename` claim for this is that we'll no longer need to do a string search across publishers -- that search isn't a huge performance hit at the moment since it only happens after JWT signature verification, but removing it would be nice!)